### PR TITLE
Drop use of deprecated filter API

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: couchdb/source/pkg/reconciler/controller.go [hunk 1]
```diff
@@ -70,12 +70,12 @@ func NewController(
 	couchdbSourceInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
 
 	deploymentInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-		FilterFunc: controller.Filter(v1alpha1.SchemeGroupVersion.WithKind("CouchDbSource")),
+		FilterFunc: controller.FilterGroupKind(v1alpha1.Kind("CouchDbSource")),
```
Comment (dprotaso) [inline couchdb/source/pkg/reconciler/controller.go]: surprised this isn't `CouchDB`

Comment (mattmoor) [reply couchdb/source/pkg/reconciler/controller.go]: Same, but that ship has sailed.

**Comment by mattmoor** [comment]
Rebased

**Pull request merged**